### ISSUE TITLE
Should URI-encoded parameter values be decoded by roaster or by application code?

### ISSUE THREAD

**Issue by ahenket** [eeditiones/roaster#25]
When you send special characters in the request string that need uri-encoding like "2013-10-15T09:34:32", you will get ```2013-10-15T09%3A34%3A32``` in your operation code.

Obviously you could ```xmldb:decode-uri()``` every parameter in your code, but since a regular xquery never has to do this, I was surprised I had to do so now.

Would it not make sense to send the uri-decoded data from the router to the code?

It saves messy code like this:

```xquery
let $effectiveDate      := xmldb:decode-uri(xs:anyURI(string($request?parameters?effectiveDate)))[string-length() gt 0]
```

**Comment by line-o** [comment]
In general I would expect roaster to handle as much as possible. So I am in favour of this change.
@ahenket can you give me an example setup (route spec and handler) for me to reproduce the issue?

**Comment by line-o** [comment]
Percent encoded characters in query parameters are decoded before they are added to request map.
With #71 this is now also true for parameters in path.

**Comment by line-o** [comment]
Included in v1.8.1
